spec/oas3.patched/components/schemas/DeviceStateFlexible.yaml

```diff
@@ -0,0 +1,3 @@
+anyOf:
+  - $ref: './DeviceState.yaml'
+  - type: string
```

Comment:
Oh.. now I get it.

Reply:
Am I missing something, or do they not generate a way to access the string (non-enum) values?

Reply:
Hmm, no, it looks like the generator doesn't successfully expose the fallback: https://github.com/equinix-labs/metal-go/pull/134/files#diff-ba37d6e9378c96645c1576db6979c216eb53740c1482bc5719d328322ee0cfb4R22

That seems like a generator bug, but an alternative approach might be to create an `EnumFallback.yaml` schema that is just a string, and reference that instead of putting `type: string` directly in the `anyOf`.

Reply:
Now that we're all-in with custom templates, a different option would be something like an `x-enum-as-string` extension:  if a property has `x-enum-as-string: true`, define the corresponding attribute as a plain old string instead of an enum.  I think we'd still want to generate the enum for convenience, though (so there'd still be, e.g., a `DeviceStatus` enum, but `Device.Status` would be a `string`, not a `DeviceStatus`.  We'd need to be careful not to use `x-enum-as-string: true` on enums on [schemas that are in `oneOf` relations](https://github.com/equinix-labs/metal-go/pull/134#issuecomment-1701582292).

Reply:
Actually looks like we could use this `anyOf` construct alongside the [`SIMPLIFY_ANYOF_STRING_AND_ENUM_STRING` generator option](https://openapi-generator.tech/docs/customization#openapi-normalizer) to do what's described in the previous comment, with the same caveat that we need to be careful to only add flexibility where it makes sense.

Reply:
While investigating fixes for #154 I found that the `SIMPLIFY_ANYOF_STRING_AND_ENUM_STRING` doesn't seem to _do_ anything; I could see from generator output that the option was recognized and accepted by the generator, but it didn't actually simplify the `anyOf` to a string.  More digging is needed to see if this is a generator bug or an issue with the combination of config options we're using.